Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -898,16 +899,37 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::splice)
         return {};
     }
 
-    auto removed_elements = Array::create(global_object);
+    // FIXME: Use ArraySpeciesCreate.
+    auto removed_elements = Array::create(global_object, actual_delete_count);
+    if (vm.exception())
+        return {};
 
     for (size_t i = 0; i < actual_delete_count; ++i) {
-        auto value = this_object->get(actual_start + i);
+        auto from = actual_start + i;
+        bool from_present = this_object->has_property(from);
         if (vm.exception())
             return {};
 
-        removed_elements->indexed_properties().append(value);
+        if (from_present) {
+            auto from_value = this_object->get(actual_start + i);
+            if (vm.exception())
+                return {};
+
+            auto define_property_return = removed_elements->define_property(i, from_value);
+            if (vm.exception())
+                return {};
+
+            if (!define_property_return) {
+                vm.throw_exception<TypeError>(global_object, ErrorType::ObjectDefinePropertyReturnedFalse);
+                return {};
+            }
```

Comment:
`define_property()` has a `throw_exceptions` argument, please use that - It'll surface the actual error, not just a generic "it returned false".

Reply:
Ah, didn't know that. Just removed all the return stuff as it defaults to throwing.